web/testing/unminified.html

```diff
@@ -42,62 +42,68 @@
       }).then(function() {
         loadKeyboards();
       });
-    </script> 
-    
+    </script>
+
     <!-- Add keyboard management script for local selection of keyboards to use -->
     <script src="commonHeader.js"></script>
-  
+
   </head>
 
-<!-- Sample page HTML -->  
-  <body> 
+<!-- Sample page HTML -->
+  <body>
     <h2>KeymanWeb Sample Page - Unminified Source</h2>
 
     <div>
-    <!-- 
+    <!--
       The following DIV is used to position the Button or Toolbar User Interfaces on the page.
       If omitted, those User Interfaces will appear at the top of the document body.
       (It is ignored by other User Interfaces.)
     -->
-    <div id='KeymanWebControl'></div>                                                                                   
+    <div id='KeymanWebControl'></div>
 
     <h3>Type in your language in this text area:</h3>
     <textarea id='ta1' class='test' placeholder='Type here'></textarea>
-  
+
     <h3>or in this input field:</h3>
     <input class='test' value='' placeholder='or here'/>
-    
+
     <!--  The following elements show how the language menu can be dynamically extended at any time -->
     <h3>Add a keyboard by keyboard name:</h3>
     <input type='input' id='kbd_id1' class='kmw-disabled' onkeypress="clickOnEnter(event,1);"/>
     <input type='button' id='btn1' onclick='addKeyboard(1);' value='Add' />
- 
+
     <h3>Add a keyboard by BCP-47 language code:</h3>
     <input type='input' id='kbd_id2' class='kmw-disabled' onkeypress="clickOnEnter(event,2);"/>
     <input type='button' id='btn2' onclick='addKeyboard(2);' value='Add' />
 
     <h3>Add a keyboard by language name:</h3>
     <input type='input' id='kbd_id3' class='kmw-disabled' onkeypress="clickOnEnter(event,3);"/>
-    <input type='button' id='btn3' onclick='addKeyboard(3);' value='Add' />   
+    <input type='button' id='btn3' onclick='addKeyboard(3);' value='Add' />
 
     <h3><a href="./">Return to testing home page</a></h3>
+
+    <h3>Control keyboard position</h3>
+    <input type='button' onclick='keyman.osk.restorePosition(true);document.getElementById("ta1").focus();' value='restorePosition(true)' />
+    <input type='button' onclick='keyman.osk.restorePosition(false);document.getElementById("ta1").focus();' value='restorePosition(false)' />
+    <input type='button' onclick='keyman.osk.restorePosition();document.getElementById("ta1").focus();' value='restorePosition()' />
+    <input type='button' onclick='keyman.osk.setRect({left:16, top:16});document.getElementById("ta1").focus();' value='setRect(16,16)' />
```

Comment:
@jahorton You may want to place this into a separate html file? I just wanted a quick and easy place to test the `restorePosition()` function

Reply:
Yeah, I'd definitely prefer a separate HTML file.  Should be simple enough to (mostly) copy and paste this file with a new name in its own folder; I did that for a lot of the test pages.